client/extensions/woocommerce/state/sites/test/request.js

```diff
@@ -0,0 +1,117 @@
+		const getResponse = { name: 'bogus get response', bogus: true };
+
+		useSandbox();
+		useNock( ( nock ) => {
+			nock( 'https://public-api.wordpress.com:443' )
+				.get( '/rest/v1.1/jetpack-blogs/123/rest-api/' )
+				.query( { path: '/wc/v3/bogus_endpoint&_method=get', json: true } )
+				.reply( 200, { data: getResponse } );
+		} );
+
+		it( 'should fetch data via promise', () => {
+			return request( siteId ).get( 'bogus_endpoint' ).then( ( data ) => {
+				expect( data ).to.eql( getResponse );
+			} );
+		} );
+
+		it( 'should catch error via promise', () => {
+			return request( siteId ).get( 'bad_bogus_endpoint' ).catch( ( error ) => {
+				expect( error.statusCode ).to.equal( 404 );
+			} );
+		} );
+	} );
+
+	describe( '#post', () => {
+		const siteId = '123';
+		const body = { name: 'bogus post request', bogus: true };
+		const postResponse = { name: 'bogus post response', bogus: true };
+
+		useSandbox();
+		useNock( ( nock ) => {
+			nock( 'https://public-api.wordpress.com:443' )
+				.post( '/rest/v1.1/jetpack-blogs/123/rest-api/' )
+				.query( { path: '/wc/v3/bogus_endpoint&_method=post', json: true } )
+				.reply( 201, { data: postResponse } );
+		} );
+
+		it( 'should post data', () => {
+			return request( siteId ).post( 'bogus_endpoint', body ).then( ( data ) => {
+				expect( data ).to.eql( postResponse );
+			} );
+		} );
+
+		it( 'should catch error via promise', () => {
+			return request( siteId ).post( 'bad_bogus_endpoint' ).catch( ( error ) => {
+				expect( error.statusCode ).to.equal( 404 );
+			} );
+		} );
+	} );
+
+	describe( '#put', () => {
+		const siteId = '123';
+		const body = { name: 'bogus put request', bogus: true };
+		const putResponse = { name: 'bogus put response', bogus: true };
+
+		useSandbox();
+		useNock( ( nock ) => {
+			nock( 'https://public-api.wordpress.com:443' )
+				.post( '/rest/v1.1/jetpack-blogs/123/rest-api/' )
+				.query( { path: '/wc/v3/bogus_endpoint&_method=put', json: true } )
+				.reply( 200, { data: putResponse } );
+		} );
+
+		it( 'should put data', () => {
+			return request( siteId ).put( 'bogus_endpoint', body ).then( ( data ) => {
+				expect( data ).to.eql( putResponse );
+			} );
+		} );
+
+		it( 'should catch error via promise', () => {
+			return request( siteId ).put( 'bad_bogus_endpoint' ).catch( ( error ) => {
+				expect( error.statusCode ).to.equal( 404 );
+			} );
+		} );
+	} );
+
+	describe( '#del', () => {
+		const siteId = '123';
+		const body = { name: 'bogus delete request', bogus: true };
+		const deleteResponse = { name: 'bogus delete response', bogus: true };
+
+		useSandbox();
+		useNock( ( nock ) => {
+			nock( 'https://public-api.wordpress.com:443' )
+				.post( '/rest/v1.1/jetpack-blogs/123/rest-api/' )
+				.query( { path: '/wc/v3/bogus_endpoint&_method=delete', json: true } )
+				.reply( 200, { data: deleteResponse } );
+		} );
+
+		it( 'should delete', () => {
+			return request( siteId ).del( 'bogus_endpoint', body ).then( ( data ) => {
```

Comment:
`DELETE` queries don't allow `body` contents. The second argument to the `.del()` function is in fact ignored.